Tw/feature/add release notes page (#69)
ecg-icas/icas: src/source/release-notes.rst

.. index:: Release Notes
.. _release_notes:

Release Notes
=============

> This section attempts to give a little more insight on what our various releases contain, as compared to previous versions.

Reviewer: ```suggestion
This section gives insight on what our current and previous releases contain.
```

Feb 2023: 'Campaigns & Micros'
------------------------------

Synopsis
""""""""

> * For bid, budgets & spent values we now report in micros: 1 cent == 10000 micros. These fields will now also be named accordingly, mentioning whether they represent a value in cents or micros. This is also represented in the categories data, where the interval of allowed values is defined. Mind, this does not include values like 'AUTOMATIC' or 'UNLIMITED'.

Reviewer: imho the word "bid" has not been introduced before to our API partners, at least not officially?

Reviewer: > Mind, this does not include values like 'AUTOMATIC' or 'UNLIMITED'.
Can you expand on this? I got confused for a second on what _this_ refers to

Author: hm, we have fields called 'bidMicros', so in a way, yes it will be introduced in this version. I can make an explicit piece about that re-naming.
I'll also rephrase this section to make it clearer.

> * Introduced campaigns, campaign ID not (yet) mandatory while placing ads. Campaigns have their own daily/total budgets. Since currently only 1 campaign is allowed per user, this effectively represents the seller's complete budget.

Reviewer: nit: we first speak of "budgets" but then "budget". Maybe rephrase a bit?

* Moved price and related fields into a struct.

> * removed externalID, use vendorID instead.

Reviewer: Maybe bundle the below into "removed fields" bullet list?

Author: bullet lists broken in this version of sphinx/docutils 😭 I can make a separate heading, sure.

* removed currency, 1 country == 1 currency anyway. Monetary values always in local currency.

> * removed pageNumber & suggestedCpcForPageOne: Numbers are unreliable and only work for the leaf category browse action, which by now is only a very small portion of our traffic on any of our marketplaces. Therefore, these numbers are not representative anymore, and we chose to remove them.

Reviewer: I would invite here a look from the product owner, to rephrase it a little bit so it's clearer and sounds less bad?

Author: I can rephrase it in context of 'traffic shape has changed since this feature was built, it's reliability has been going down over time' ?

Author: will ask!

   - removed
 * - suggestedCpcForPageOne
   - removed
 * - allowPaypal            
   - removed

Categories
""""""""""

See :ref:`categories` for full details.

.. list-table::
 :widths: 20 80
 :header-rows: 1

 * - Field
   - Change synopsis
 * - currency               
   - removed
 * - cpc
   - moved to bidMicros.
 * - totalBudget            
   - moved to totalBudgetMicros.
 * - minDailyBudget
   - moved to dailyBudgetMicros. This is now an interval instead of min value.
 * - paypalEnabled
   - removed
 * - paypalBpEnabled
   - removed
 * - suggestedCpcForPageOne
   - removed

Campaigns
"""""""""
Added first campaign support. See :ref:`campaigns` for full details.

* :ref:`get_campaign` to get list of campaigns for a user. Expected 0 or 1 result.
* :ref:`get_campaign_id` to fetch the individual campaign.
* :ref:`post_campaign` to create a campaign. Currently only 1 allowed per seller.
* :ref:`put_campaign_id` to update the campaign. Currently only status & budgets can be updated.
* :ref:`put_campaign_id_status` to manage the campaign status (ACTIVE,PAUSED) directly.
* :ref:`put_campaign_id_budgets` to manage the campaign's budgets directly.

.. note::

>  For the time being, we will create a campaign underwater for users who start placing ads without first having created a campaign. 

Reviewer: ```suggestion
 For the time being, we will create a campaign underwater for users who start placing ads without first having created a campaign. All subsequent ads will belong to that default campaign.
```

 Be sure to check the budgets of that campaign as a default campaign will have unlimited budget.
 Deleting a campaign is currently not allowed.

User
""""

See :ref:`get_user` for full details.

Start using V4. There are 2 new fields in comparison with V2:
``hasAds`` (**bool**), whether a user has ads in the system and 

> ``isApiManaged`` (**bool**), whether we see API partners managing ads for the user (not true if an api partner is using RO scopes only)

Reviewer: I'm not sure why we decided to expose it in the end :-/ it was supposed to be a temporary thing, given the implementation is based also on "proxy" data. It could be that the user is API managed but not much activity in the past (90?) days so we see no active tokens

Author: if we want to have only 1 user version available, we need to be able to explain all the fields. I'd also rather remove these two fields, but 
1) console needs it 
2) BNL wants to keep the functionality around `isAPiManaged` since it's about an upsell message 😭
